File: WebApplication2/WebApplication2/MyBootStrapper.vb
Imports Nancy.Bootstrapper
Imports Nancy.Authentication.Forms
Imports Nancy.Hosting.Aspnet
Imports Nancy
Imports Nancy.TinyIoc
Imports WebApplication2.Security
Imports WebApplication2.Services

Public Class MyBootStrapper
    Inherits DefaultNancyBootstrapper

    Protected Overrides Sub ConfigureApplicationContainer(container As TinyIoCContainer)
        container.Register(Of BushService).AsSingleton()
        container.Register(Of TreeService).AsSingleton()
        container.Register(Of UserService).AsSingleton()
    End Sub

    Protected Overrides Sub ConfigureRequestContainer(container As TinyIoCContainer, context As NancyContext)
        MyBase.ConfigureRequestContainer(container, context)
        container.Register(Of IUserMapper, FakeUserMapper)()
    End Sub

    Protected Overrides Sub RequestStartup(container As TinyIoCContainer, pipelines As IPipelines, context As NancyContext)
        Dim formsAuthConfiguration = New FormsAuthenticationConfiguration With
        {
            .RedirectUrl = "~/login",
            .UserMapper = container.Resolve(Of IUserMapper)()
        }
        FormsAuthentication.Enable(pipelines, formsAuthConfiguration)
    End Sub
End Class

File: WebApplication2/WebApplication1/MyBootStrapper.cs
using Nancy;
using Nancy.TinyIoc;
using WebApplication1.Services;

namespace WebApplication1
{
    public class MyBootStrapper : DefaultNancyBootstrapper
    {
    

        protected override void ConfigureApplicationContainer(TinyIoCContainer container)
        {
            container.Register<BushService>().AsSingleton();
            container.Register<TreeService>().AsSingleton();
        }
    }
}
